go/cmd/rulesctl/cmd/add_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+	"os"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAdd(t *testing.T) {
+	cmd := Add()
+	require.NotNil(t, cmd)
+	require.Equal(t, "add-rule", cmd.Name())
+	configFile = "./testdata/rules.json"
+
+	tests := []struct {
+		name           string
+		args           []string
+		expectedOutput string
+	}{
+		{
+			name:           "Action fail",
+			args:           []string{"--dry-run=true", "--name=Rule", `--description="New rules that will be added to the file"`, "--action=fail", "--plan=Select"},
+			expectedOutput: "[\n  {\n    \"Description\": \"Some value\",\n    \"Name\": \"Name\",\n    \"Action\": \"FAIL\"\n  },\n  {\n    \"Description\": \"\\\"New rules that will be added to the file\\\"\",\n    \"Name\": \"Rule\",\n    \"Plans\": [\n      \"Select\"\n    ],\n    \"Action\": \"FAIL\"\n  }\n]\n",
```

Comment:
imo these strings will be much easier to read and maintain if we use the backtick style, e.g.:

```go
expectedOutput: `[
    {
        "Description": "Some value",
        "Name": "Name",
        ...
    }
]`
```